src/haddock/libs/libworkflow.py

```diff
@@ -26,7 +26,7 @@ def __init__(self, workflow_params, start=0, **other_params):
         # terminate is used to synchronize the `clean` option with the
         # `exit` module. If the `exit` module is removed in the future,
         # you can also remove and clean the `terminate` part here.
-        self._terminated = 0
+        self._terminated = None
```

Comment:
Note for the future. This was one of the most tricky parts of this implementation.